### PR TITLE
Initial commit, added Coco2017 dataset with Panoptic annotations

### PR DESCRIPTION
**Pull request by kaustumbh7**
Fixes https://github.com/tensorflow/datasets/issues/25

Signed-off-by: Kaustumbh <kaustumbhjaiswal11@gmail.com>

### CODE REVIEW

**Comment by kaustumbh7** [comment]
I signed it! 😊

File: tensorflow_datasets/image/coco2017_panoptic.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
+from __future__ import absolute_import
+from __future__ import division
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/coco2017_panoptic.py]: Would it be possible to add tests eventually ?

File: tensorflow_datasets/image/coco2017_panoptic.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+            "annotations.\n"
+            " * The test split don't have any annotations (only images).\n"
+            " * Coco2017 defines 133 classes.\n"),
+        features=tfds.features.FeaturesDict({
+            # Images can have variable shape
+            "image": tfds.features.Image(encoding_format="jpeg"),
+            "image/filename": tfds.features.Text(),
+            "panoptic_image": tfds.features.Image(encoding_format="png"),
+            "panoptic_image/filename": tfds.features.Text(),
+            "segments_info": tfds.features.SequenceDict({
```
[the remark this answers is not in the document]
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: Yeah, sure. I would do that.

File: tensorflow_datasets/image/coco2017_panoptic.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+                "area": tf.int64,
+                "id": tf.int64,
+                "bbox": tfds.features.BBoxFeature(),
+                # Coco2017 has 133 categories
```
[the remark this answers is not in the document]
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: I am not sure of that. There are 133 categories in total. I'll look into it.

File: tensorflow_datasets/image/coco2017_panoptic.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+            "captioning dataset. This version contains images, panoptic segmentation "
+            "and labels for the 2017 version.\n"
+            "Note:\n"
+            " * Some images from the train and validation sets don't have "
+            "annotations.\n"
+            " * The test split don't have any annotations (only images).\n"
+            " * Coco2017 defines 133 classes.\n"),
+        features=tfds.features.FeaturesDict({
+            # Images can have variable shape
+            "image": tfds.features.Image(encoding_format="jpeg"),
```
[the remark this answers is not in the document]
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: Sure!

File: tensorflow_datasets/image/coco2017_panoptic.py [hunk 5]
```diff
@@ -0,0 +1,263 @@
+from __future__ import absolute_import
```
[the remark this answers is not in the document]
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: I'll revise the code and see if it is possible to inherit some of the functions from Coco2014.

File: tensorflow_datasets/image/coco2017_panoptic.py (hunk 1 above)
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: I have to generate tests as per the documentation, right?

Comment (cyfra) [reply tensorflow_datasets/image/coco2017_panoptic.py]: Yes, please add tests - without them we'll not be able to detect issues with your code in the future.

Most of the work is actually in creating the fake data, the test file itself should be the easier part (please look at the documentation for details).

Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: Okay, I'll do that.

Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: @Conchylicultor Can you please help me ? I am getting this error when running test cases- 
```
.Downloading / extracting dataset coco2017 (?? GiB) to /var/folders/hr/dm5hrrkx1cq_h5kxx8h_k21c0000gn/T/coco2017_panoptic_testlW2lBH/tmpl1iUzS/coco2017/1.0.0...
E..s
======================================================================
ERROR: test_download_and_prepare_as_dataset (__main__.Coco2017_PANOPTIC_TEST)
test_download_and_prepare_as_dataset (__main__.Coco2017_PANOPTIC_TEST)
Run the decorated test method.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/absl/third_party/unittest3_backport/case.py", line 37, in testPartExecutor
    yield
  File "/Library/Python/2.7/site-packages/absl/third_party/unittest3_backport/case.py", line 162, in run
    testMethod()
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/testing/test_utils.py", line 167, in decorated
    f(self, *args, **kwargs)
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/testing/dataset_builder_testing.py", line 194, in test_download_and_prepare_as_dataset
    self._download_and_prepare_as_dataset(self.builder)
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/testing/dataset_builder_testing.py", line 209, in _download_and_prepare_as_dataset
    builder.download_and_prepare(download_config=download_config)
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/core/api_utils.py", line 52, in disallow_positional_args_dec
    return fn(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 219, in download_and_prepare
    max_examples_per_split=download_config.max_examples_per_split)
  File "/Library/Python/2.7/site-packages/tensorflow_datasets/core/dataset_builder.py", line 650, in _download_and_prepare
    for split_generator in self._split_generators(dl_manager):
  File "/Users/KJ/Desktop/GitHub/datasets_tf/tensorflow_datasets/image/coco2017_panoptic.py", line 97, in _split_generators
    image_dir=extracted_paths["train_images"],
KeyError: 'train_images'

----------------------------------------------------------------------
Ran 5 tests in 0.566s

FAILED (errors=1, skipped=1)
```

Comment (Conchylicultor) [reply tensorflow_datasets/image/coco2017_panoptic.py]: You probably need to add `DL_EXTRACT_RESULT` in the test, like in https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/image/mnist_test.py

**Comment by hadim** [comment]
Any reason why the masks are not included for each bboxes when available from coco? I guess the same question apply to coco2014.

**Comment by Conchylicultor** [comment]
> Any reason why the masks are not included for each bboxes when available from coco? I guess the same question apply to coco2014.

For Coco2014, the `pycocotools` mask lib did not seems to work with python3, so we haven't added the masks in the first version.

**Comment by hadim** [comment]
I used pycocotools successfully in the past with Python 3. I even created the conda forge package: https://github.com/conda-forge/pycocotools-feedstock

Interested by a PR? It might take some time from my side so if @kaustumbh7 you are willing to make one, feel free.

**Comment by jpuigcerver** [comment]
What's the status of this CL? What's preventing it from being pushed?

File: tensorflow_datasets/image/coco2017_panoptic.py (hunk 1 above)
Comment (kaustumbh7) [reply tensorflow_datasets/image/coco2017_panoptic.py]: @Conchylicultor Okay. I''l try it out.

**Comment by Conchylicultor** [comment]
> What's the status of this CL? What's preventing it from being pushed?

@jpuigcerver We would like to have tests for this dataset, to ensure it is generating correctly.

**Pull request closed without merging**